src/Worker.Extensions.DurableTask/HTTP/HttpMethodConverter.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.DurableTask.Http;
+
+internal class HttpMethodConverter : JsonConverter<HttpMethod>
+{
+    public override bool CanConvert(Type objectType)
+    {
+        return typeof(HttpMethod).IsAssignableFrom(objectType);
+    }
+
+    public override HttpMethod Read(
+        ref Utf8JsonReader reader,
+        Type objectType,
+        JsonSerializerOptions options)
+    {
+        HttpMethod method = HttpMethod.Get;
+
+        if (reader.TokenType != JsonTokenType.StartObject)
```

Comment:
This doesn't look right. I wouldn't expect a start-object here at all, it is going to be a literal string token. Implementation should be a pure `return new HttpMethod(reader.GetString());`. No need to perform checks ahead. If it is `null`, this won't be called. If this isn't a string, `.GetString()` will throw (I think), which works for us.